lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -47,8 +65,48 @@ def test_fig_close(backend):
     assert init_figs == Gcf.figs
 
 
+@pytest.mark.backend('Qt5Agg')
+@pytest.mark.parametrize("target, kwargs, windows_left", [
+    (plt.show, {"block": True}, 0),
+    (plt.pause, {"interval": 10}, 1)
+])
+def test_sigint(qt_core, platform_simulate_ctrl_c, target,
+                kwargs, windows_left):
+    plt.ion()
+    fig, ax = plt.subplots()
+    xdata, ydata = [], []
+    ln, = plt.plot([], [], 'ro')
+
+    def init():
+        ax.set_xlim(0, 100)
+        ax.set_ylim(0, 100)
+        return ln,
+
+    def update(frame):
+        if frame < 50:
+            xdata.append(frame)
+            ydata.append(frame)
+            ln.set_data(xdata, ydata)
+            return ln,
+        elif frame == 50:
+            platform_simulate_ctrl_c()
+            return ln,
+        else:
+            return ln,
+
+    ani = FuncAnimation(fig, update, frames=range(0, 100),
```

Comment:
This is a super clever way of injecting a SIGINT!  However, it may be simpler to use a single-shot timer.